libs/langchain/langchain/llms/watsonxllm.py

```diff
@@ -0,0 +1,327 @@
+                    get_from_dict_or_env(values, "instance_id", "WATSONX_INSTANCE_ID")
+                )
+
+        try:
+            from ibm_watson_machine_learning.foundation_models import Model
+
+            credentials = {
+                "url": values["url"].get_secret_value() if values["url"] else None,
+                "apikey": values["apikey"].get_secret_value()
+                if values["apikey"]
+                else None,
+                "token": values["token"].get_secret_value()
+                if values["token"]
+                else None,
+                "password": values["password"].get_secret_value()
+                if values["password"]
+                else None,
+                "username": values["username"].get_secret_value()
+                if values["username"]
+                else None,
+                "instance_id": values["instance_id"].get_secret_value()
+                if values["instance_id"]
+                else None,
+                "version": values["version"].get_secret_value()
+                if values["version"]
+                else None,
+            }
+            credentials_without_none_value = {
+                key: value for key, value in credentials.items() if value is not None
+            }
+
+            watsonx_model = Model(
+                model_id=values["model_id"],
+                credentials=credentials_without_none_value,
+                params=values["params"],
+                project_id=values["project_id"],
+                space_id=values["space_id"],
+                verify=values["verify"],
+            )
+            values["watsonx_model"] = watsonx_model
+
+        except ImportError:
+            raise ImportError(
+                "Could not import ibm_watson_machine_learning python package. "
+                "Please install it with `pip install ibm_watson_machine_learning`."
+            )
+        return values
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            "model_id": self.model_id,
+            "params": self.params,
+            "project_id": self.project_id,
+            "space_id": self.space_id,
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "IBM watsonx.ai"
+
+    def _create_llm_result(self, response: Any) -> LLMResult:
+        """Create the LLMResult from the choices and prompts."""
+        generations = []
+        for res in response:
+            finish_reason = res.get("results")[0].get("stop_reason")
+            gen = Generation(
+                text=res.get("results")[0].get("generated_text"),
+                generation_info={
+                    "generated_token_count": res.get("results")[0].get(
+                        "generated_token_count"
+                    ),
+                    "input_token_count": res.get("results")[0].get("input_token_count"),
```

Comment:
I would put this under the `token_usage` property. This is standard - look at OpenAI implementation.